unitTests/ui/InitializationTest.ts

```diff
@@ -399,6 +399,73 @@ export function InitializationTest() {
       Simulate.removeJQuery();
     });
 
+    describe('when initializing a Search Interface', () => {
+      let spyAfterComponentsInitialized;
+      const doInit = () => {
+        return Initialization.initializeFramework(root, searchInterfaceOptions, () => {
+          const initResult = Initialization.initSearchInterface(root, searchInterfaceOptions);
+          return initResult;
+        });
+      };
+      beforeEach(() => {
+        spyAfterComponentsInitialized = jasmine.createSpy('afterComponentsInitialized');
+        $$(root).on(InitializationEvents.afterComponentsInitialization, spyAfterComponentsInitialized);
+      });
+      afterEach(() => {
+        spyAfterComponentsInitialized = null;
+      });
+      it('will trigger the afterComponentsInitialized event handler', async done => {
+        await doInit();
+        expect(spyAfterComponentsInitialized).toHaveBeenCalled();
+        done();
+      });
+      describe('when afterComponentsInitialized is deferred by a component', () => {
+        let deferSpy;
+        beforeEach(() => {
+          deferSpy = jasmine.createSpy('initializationSpy');
+          $$(root).on(InitializationEvents.afterComponentsInitialization, (event, data: IInitializationEventArgs) => {
+            data.defer.push(
+              new Promise(resolve => {
+                deferSpy();
+                resolve();
+              })
+            );
+          });
+        });
+        afterEach(() => {
+          deferSpy = null;
+        });
+        it('will trigger the deferred promise', async done => {
+          await doInit();
+          expect(deferSpy).toHaveBeenCalled();
+          done();
+        });
+      });
+      describe('when afterComponentsInitialized is deferred by a failing promise', () => {
+        let deferSpy;
+        beforeEach(() => {
+          deferSpy = jasmine.createSpy('initializationSpy');
+          $$(root).on(InitializationEvents.afterComponentsInitialization, (event, data: IInitializationEventArgs) => {
+            data.defer.push(Promise.reject('It failed badly!'));
+            data.defer.push(
+              new Promise(resolve => {
+                deferSpy();
+                resolve();
+              })
+            );
+          });
+        });
+        afterEach(() => {
+          deferSpy = null;
+        });
+        it('will skip the rejected promise and wait for the working one', async done => {
```

Comment:
Should we add a test for "all failing promises" ?

Should we also have a test that check that it actually waits properly that all the promises have resolved before continuing executing the rest of the initialization event ?

And maybe an explicit test that looks that if the `defer` argument is not populated, then it does not wait at all and will resolve in a synchronous manner (the "backward compatibility" check we talked about) ?